distribution/integration/tests-integration/user-core/src/test/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManagerTest.java

```diff
@@ -0,0 +1,285 @@
+
+        return new ReadOnlyLDAPUserStoreManager(realmConfig, properties, claimManager, null, null, 0);
+    }
+
+    private RealmConfiguration getRealmConfiguration() {
+        RealmConfiguration realmConfig = new RealmConfiguration();
+        realmConfig.getUserStoreProperties().put(LDAPConstants.CONNECTION_URL, "ldap://localhost:" + LDAP_SERVER_PORT);
+        realmConfig.getUserStoreProperties().put(LDAPConstants.DNS_URL, null);
+        realmConfig.getUserStoreProperties().put("AnonymousBind", "false");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.CONNECTION_NAME, "uid=admin,ou=system");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.CONNECTION_PASSWORD, "secret");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.USER_SEARCH_BASE, SEARCH_BASE);
+        realmConfig.getUserStoreProperties().put(LDAPConstants.USER_NAME_LIST_FILTER, "(objectClass=person)");
+        realmConfig.getUserStoreProperties()
+                .put(LDAPConstants.USER_NAME_SEARCH_FILTER, "(&(objectClass=person)(sn=?))");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.USER_NAME_ATTRIBUTE, "uid");
+        realmConfig.getUserStoreProperties().put(UserCoreConstants.RealmConfig.READ_GROUPS_ENABLED, "true");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.GROUP_SEARCH_BASE, "ou=Groups,dc=wso2,dc=org");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.GROUP_NAME_LIST_FILTER, "ou=Groups,dc=wso2,dc=org");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.ROLE_NAME_FILTER, "(&(objectClass=groupOfNames)(cn=?))");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.GROUP_NAME_ATTRIBUTE, "cn");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.MEMBERSHIP_ATTRIBUTE, "member");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.USER_ENTRY_OBJECT_CLASS, "identityPerson");
+        realmConfig.getUserStoreProperties().put(LDAPConstants.GROUP_ENTRY_OBJECT_CLASS, "groupOfNames");
+        return realmConfig;
+    }
+
+    private JdbcDataSource createDatasource() throws Exception {
+        JdbcDataSource dataSource = new JdbcDataSource();
+        dataSource.setURL("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1");
+        dataSource.setUser("sa");
+        dataSource.setPassword("sa");
+
+        String dbScript = this.getClass().getResource("/dbscripts/h2-um.sql").getFile();
+        loadData(dataSource, dbScript);
+        return dataSource;
+    }
+
+    private void loadData(JdbcDataSource ds, String location) throws Exception {
+        File file = new File(location);
+
+        final String LOAD_DATA_QUERY = "RUNSCRIPT FROM '" + file.getCanonicalPath() + "'";
+
+        Connection connection = null;
+        connection = ds.getConnection();
+        Statement statement = connection.createStatement();
+        statement.execute(LOAD_DATA_QUERY);
+        connection.close();
```

Comment:
Should this go to the finally block?

Reply:
No real value in "finally{close();}", as the connection and the database is discarded after every call. Hence opened connections make no issue. The database is also in-memory DB, no network calls.

Reply:
Is the database discard after every call to this method?

Reply:
This is done before each test suite runs, failing which the entire test fails. and database is discarded.